### PR TITLE
User endpoints

### PR DESCRIPTION
**Pull request by lNamelessl**
@Grey-A

### CODE REVIEW

File: .vscode/settings.json
Comment (Grey-A) [inline .vscode/settings.json:1]: kilode, what are they for

File: app/config/settings.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Settings(BaseSettings):
     ACCESS_TOKEN_EXPIRE_MINUTES: int = os.environ.get("ACCESS_TOKEN_EXPIRE_MINUTES")
     REFRESH_TOKEN_EXPIRE_HOURS: int = os.environ.get("REFRESH_TOKEN_EXPIRE_MINUTES")
     REFRESH_TOKEN_EXPIRE_HOURS_LONG: int = os.environ.get("REFRESH_TOKEN")
-
+    # PASSWORD_RESET_TOKEN_EXPIRE_MINUTES: int = os.environ.get("PASSWORD_RESET_TOKEN_EXPIRE_MINUTES")
```
Comment (Grey-A) [inline app/config/settings.py:24]: why did you comment out

File: app/user/apis.py [hunk 2]
```diff
@@ -93,3 +94,215 @@ async def user_edit(
...
+    response_model=ResponseSchema,
+)
+async def user_refresh_token(
+    db: DatabaseSession,
+    refresh_token: str = Body(
+        description="The user's refresh token", min_length=1, embed=True
+    ),
+):
+    """This endpoint generates a new access token for the user using the refresh token"""
+    if user_id := security.verify_user_refresh_token(token=refresh_token):
```
Comment (Grey-A) [inline app/user/apis.py]: this will always return the id, so no need

**Review by Grey-A — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (lNamelessl) [reply .vscode/settings.json:1]: How did they get there? Must be when I tried installing some libraries

File: app/config/settings.py (hunk 1 above)
Comment (lNamelessl) [reply app/config/settings.py:24]: Cuz it's not needed yet and it was the reason the build was failing earlier.

File: app/user/apis.py (hunk 2 above)
Comment (lNamelessl) [reply app/user/apis.py]: Okay

File: app/config/settings.py (hunk 1 above)
Comment (Grey-A) [reply app/config/settings.py:24]: add it to the pytest.ini

Comment (Grey-A) [reply app/config/settings.py:24]: add it to the pytest.ini

Comment (Grey-A) [reply app/config/settings.py:24]: add it to the `pytest.ini`

File: app/user/schemas/create_schemas.py
Comment (Grey-A) [inline app/user/schemas/create_schemas.py:1]: format this with black

File: app/user/schemas/response_schemas.py
Comment (Grey-A) [inline app/user/schemas/response_schemas.py:1]: format with black

File: app/user/services.py [hunk 3]
```diff
@@ -160,3 +160,53 @@ async def edit_user(user_id: int, data: edit_schemas.UserEdit, db: Session):
...
+            detail=f"email {email} already subscribed",
+        )
+    obj = models.NewsLetter(email=email)
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def create_company(user_id: int, data: create_schemas.CompanyCreate, db: Session):
```
Comment (Grey-A) [inline app/user/services.py]: a user can only have one company attached, make sure you validate that

File: alembic/versions/7cdda97d045f_2024_06_05-created_user_password_reset_token_table.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+down_revision: Union[str, None] = "c5d0b8e1fb42"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.create_table(
+        "user_password_reset_tokens",
+        sa.Column(
+            "id", sa.Integer, primary_key=True, autoincrement=True, nullable=False
```
Comment (Grey-A) [inline alembic/versions/7cdda97d045f_2024_06_05-created_user_password_reset_token_table.py]: Pks aren't nullable by default

File: app/user/apis.py
Comment (Grey-A) [inline app/user/apis.py:1]: Format the document using black

File: app/user/apis.py [hunk 5]
```diff
@@ -93,3 +95,210 @@ async def user_edit(
...
+    response_model=ResponseSchema,
+)
+async def user_refresh_token(
+    db: DatabaseSession,
+    refresh_token: str = Body(
+        description="The user's refresh token", min_length=1, embed=True
+    ),
+):
+    """This endpoint generates a new access token for the user using the refresh token"""
+    user_id = security.verify_user_refresh_token(token=refresh_token)
```
Comment (Grey-A) [inline app/user/apis.py:118]: Call password refresh selector first

File: app/user/selectors.py [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
+from app.config.settings import get_settings
 from app.user import models, security
 
-
 settings = get_settings()
 
 
 async def get_user_by_id(user_id: int, db: Session, raise_exception: bool = True):
     """This function returns a user obj using it's id
+    F
```
Comment (Grey-A) [inline app/user/selectors.py]: This F

File: app/user/selectors.py [hunk 7]
```diff
@@ -98,3 +98,27 @@ async def get_current_user(
...
+
+    Args:
+        user_id (int): The user's ID
+        token (str): The user's refresh token
+        db (Session): The database session
+
+    Returns:
+        (models.UserRefreshToken): The user refresh token obj
+    """
+    obj = (
```
Comment (Grey-A) [inline app/user/selectors.py]: Use a walrus operator I.e if obj := ....:
    return obj 
Raise exception

**Review by Grey-A — CHANGES_REQUESTED** [review]

File: app/user/apis.py (hunk 5 above)
Comment (Grey-A) [reply app/user/apis.py:118]: This one

Comment (lNamelessl) [reply app/user/apis.py:118]: Why? is it still not going to work the same way? They are all dependent on the userId i get from the verify_token and the selector won't get any id if it comes first

**Comment by Grey-A** [comment]
Logic, look at what the functions do

**Pull request merged**